app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -79,10 +87,67 @@ def package_version
         JSON.parse(package_json.read)['version']
       end
 
+      def publish_config
+        JSON.parse(package_json.read)['publishConfig']
+      end
+
+      def publish_config_access
+        publish_config['access'] || PRIVATE
+      end
+
+      def package_name
+        JSON.parse(package_json.read)['name']
+      end
+
+      def scoped_package?
+        package_name.start_with?('@shopify')
+      end
+
+      def valid_publish_config?
+        return true if ENV['ENFORCE_PUBLISH_CONFIG'].nil?
+
+        return false if ENV['ENFORCE_PUBLISH_CONFIG'] && publish_config.blank?
+        valid_publish_config_access? && package_scoped_when_private?
+      end
+
+      def valid_publish_config_access?
+        VALID_ACCESS.include?(publish_config_access)
+      end
+
+      # ensure private packages are scoped
+      def package_scoped_when_private?
```

Comment:
I still don't know what this means.  It makes it very difficult to understand the the descriptions.

Reply:
Okay, renamed to `private_scoped_package`